### PR TITLE
[feat-779] added generic bar chart component and changed data handlin…

### PR DESCRIPTION
**Pull request by chenmu10**
…g in widgets

### CODE REVIEW

**Comment by chenmu10** [comment]
had issues with the previous branch so created a second one.
details on the feature:
- https://github.com/hasadna/anyway-newsflash-infographics/pull/806
- https://github.com/hasadna/anyway-newsflash-infographics/issues/779

pending from backend @BarVolunteering :
1. data order in series should be consistent.
2. data should be ordered by severity 
3. label_key should be a string and not a number (for example the year '2017')

image

**Comment by BarVolunteering** [comment]
From what i know there are 3.5 widgets that need to be change:
*killed_and_injured_count_per_age_group_stacked/killed_and_injured_count_per_age_group - WIP
*accident_count_by_accident_year
*injured_count_by_accident_year

@chenmu10 Do we have other widgets that need to be change?

**Comment by chenmu10** [comment]
also:
- accident_count_by_car_type
- injured_accidents_with_pedestrians

**Comment by BarVolunteering** [comment]
@chenmu10 
Widgets:
killed_and_injured_count_per_age_group_stacked
killed_and_injured_count_per_age_group

are done.

**Comment by BarVolunteering** [comment]
As we talked in Slack - Some is done some is still in progress.

File: src/components/molecules/GenericBarChartView.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+const SingleBarChart: FC<ISingleBarChartProps> = ({ data, isPercentage, textLabel }) => {
+  const yLabels = Object.keys(data[0]);
+  yLabels.splice(0, 1);
+
+  const barStyle = {
+    filter: `drop-shadow(1mm 1mm 0 ${tinycolor(roseColor).darken().toString()})`,
+  };
+  return (
+    <>
```
Comment (yuvalbl) [inline src/components/molecules/GenericBarChartView.tsx:83]: remove

File: src/components/molecules/GenericBarChartView.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    </>
+  );
+};
+
+const MultiBarChart: FC<IMultiBarChartProps> = ({ data, isPercentage, isStacked, textLabel }) => {
+  const yLabels = Object.keys(data[0]);
+  yLabels.splice(0, 1);
+  const maxBarsNum = yLabels.length;
+  return (
+    <>
```
Comment (yuvalbl) [inline src/components/molecules/GenericBarChartView.tsx:98]: remove

File: src/utils/barchart.utils.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+type stringNumObject = Record<string, string | number>;
+type stringObject = Record<string, string>;
+
+export const transformItems = (data: any) => {
+  const isSingleBar = data.items[0].series == null;
```
Comment (yuvalbl) [inline src/utils/barchart.utils.ts:5]:
```
  const isSingleBar = data.items?.length > 0 && data.items[0].series;
```

**Comment by chenmu10** [comment]
We will split the tasks. 
only the new component addition is here: https://github.com/hasadna/anyway-newsflash-infographics/pull/849

**Pull request closed without merging**